app/views/admin/pages/index.html.erb

```diff
@@ -44,3 +44,57 @@
 	<% end %>
 	<% end %>
 </table>
+
+<% if @there_are_hidden_pages %>
+<h2>
+	Hidden Pages
+</h2>
+
+<table class="list">
+	<tr>
+		<th class="main">
+			Page
+		</th>
+		<th>
+			Actions
+		</th>
+	</tr>
+	<% if @tl_hidden_pages.count.positive? %>
```

Comment:
Admittedly this is a minor premature optimization below, but you might also prefer the last option using `#any?` for readability, and treat the optimization as a bonus:

```rb
# #count will *always* perform a SQL COUNT query:
<% if @tl_hidden_pages.count.positive? %>
```
```rb
# Using #size instead of #count does one of:
#
# 1) If @tl_hidden_pages is NOT loaded (i.e. no SELECT * query done yet),
# then a SQL COUNT query is performed.
#
# 2) If @tl_hidden_page IS loaded (i.e. SELECT * query done),
# then no SQL COUNT query is performed, instead @tl_hidden_pages.length
# (Array#length) is called, saving a trip to the DB.
<% if @tl_hidden_pages.size.positive? %>
```
```rb
# Using #any? instead performs a similar optimization to
# #size doing one of:
#
# 1) If @tl_hidden_pages is NOT loaded (i.e. no SELECT * query done yet),
# then a SQL SELECT query is performed.
#
# 2) If @tl_hidden_page IS loaded (i.e. SELECT * query done),
# then no SQL SELECT query is performed, instead !@tl_hidden_pages.empty?
# (!Array#empty?) is called, saving a trip to the DB.
# 
# (also shortens the method chain by one which is nice for readability)
<% if @tl_hidden_pages.any? %>
```

If you ever want to experiment to see the differences in what SQL is run:

```rb
bin/rails console

# Enable SQL logging in console:
ActiveRecord::Base.logger = Logger.new(STDOUT); nil

# Need `; nil` on end of line to prevent console from calling tl_hidden_pages.inspect
# and prematurely querying the DB:
tl_hidden_pages = Page.top_level_hidden_pages; nil

tl_hidden_pages.count # should perform SQL COUNT

tl_hidden_pages.size # should perform SQL COUNT

tl_hidden_pages.any? # should perform SQL SELECT

# Call a method that would load tl_hidden_pages and perform SELECT * query,
# e.g.:
tl_hidden_pages.load
# tl_hidden_pages.each(&:id) # looping would also perform SELECT * query.

tl_hidden_pages.size # NO SQL COUNT query
tl_hidden_pages.any? # NO SQL SELECT query
tl_hidden_pages.count # performs (unnecessary) SQL COUNT query
```

Reply:
Neat!